cdci_data_analysis/analysis/parameters.py

```diff
@@ -167,24 +167,40 @@ def __init__(self,
                  value=None,
                  units=None,
                  name: Union[str, None]=None, 
-                 allowed_units=[],
+                 allowed_units=None,
                  default_units=None,
+                 default_type=None,
+                 allowed_types=None,
                  check_value=None,
                  allowed_values=None,
                  units_name=None):
 
+        if (units is None or units == '') and \
+                default_units is not None and default_units != '':
+            # TODO ideally those should be renamed as singular (unit and default_unit)
+            #  but they are as such because they're used also in plugins
+            #
+            units = default_units
+
         self.check_value = check_value
 
         if allowed_units is None:
             allowed_units = []
         else:
             allowed_units = allowed_units.copy()
 
+        if allowed_types is None:
+            allowed_types = []
```

Comment:
ok, then we can follow the same approach with `allowed_units` ?

Reply:
and further reduce some code

Reply:
also, the code was designed with the approach of performing a unit(s) check only in case the list of allowed units was not None and not empty:

[https://github.com/oda-hub/dispatcher-app/blob/b1773a866332301b0be602bbf16238e3f0aa7e78/cdci_data_analysis/analysis/parameters.py#L237-L239](url)

Reply:
> also, the code was designed with the approach of performing a unit(s) check only in case the list of allowed units was not None and not empty:
> 
> [https://github.com/oda-hub/dispatcher-app/blob/b1773a866332301b0be602bbf16238e3f0aa7e78/cdci_data_analysis/analysis/parameters.py#L237-L239](url)

I saw that it ignores `[]` as well as `None`, but why make two indicators for disabled check? 


> ok, then we can follow the same approach with `allowed_units` ?

we can!

Reply:
> but why make two indicators for disabled check?

nice question, one of those parts of the code that I would definitely modify if it wasn't for the fact that I would be concerned of breaking something else. somewhere else

Reply:
you can modify it by preserving interface. E.g. when units are set - `allowed_units` can be checked for `[]`, and if so - assigned `None`. And a warning given.

This is a relatively small part of code, all contained in one 700-line file. But it's old, written with little attention, and affects lot's of things. 
Doing it right will improve on various current and future issues with parameter types.